app/uk/gov/hmrc/cataloguefrontend/CatalogueController.scala

```diff
@@ -305,18 +306,11 @@ class CatalogueController @Inject()(
             envName -> deployments.filter(_.environmentMapping.name == envName)
           }.toMap
 
-          /*
-           * Note that we ignore libraryDependencies obtained from parsing master / GitHub, and instead use those
-           * from the 'latest' slug.  As such, 'Platform Dependencies' on all tabs (including 'Latest') is populated
-           * from slug info.
-           * Slugs do not contain build related information however.  So 'Build Tools' continues to display information
-           * parsed from master / GitHub.
-           */
           Ok(
             serviceInfoPage(
               repositoryDetails.copy(environments = optDeployedEnvironments, jenkinsURL = jenkinsLink),
-              optMasterDependencies.map(_.copy(libraryDependencies = librariesOfLatestSlug)),
-              librariesBySlugVersion,
+              optMasterDependencies,
+              librariesBySlugVersion + (Latest.asString -> librariesOfLatestSlug),
```

Comment:
Can `serviceInfoPage` take `SlugInfoFlag` as key rather than String?
(Similarly deploymentsByEnvironmentName?)

Reply:
Most of the slugs are being loaded by an explicit deployment version number.  However, we now also need to load "latest" - which uses a different API.  That API uses `SlugInfoFlag` - but we only care about one instance of those flags - `Latest`.  Essentially our key should be an explicit `Version` or some indicator of "latest version".

At the service level I introduced a wrapper algebraic type to capture this.
However, this hasn't been brought into the UI - because the UI seemed to be mostly String-typed - and it felt like it would be at odds with the existing code.

I didn't namespace `SlugInfoFlag.Latest` as this code doesn't really reference `SlugInfoFlag`.  We are just using the constant String "latest" to distinguish this from something like "1.2.3" - it could have been any String as long as it does not match a real version - and it just so happens that the constant that seemed most sensible was already defined.

I agree that this is only possible because the key is a String, and that it does feel "dirty".
The page currently treats "latest" specially anyway.  The data was added in this way simply to avoid adding another parameter - but it could just as easily have been passed separately - leaving `librariesBySlugVersion` to just contain the explicitly versioned slugs.  I'm happy to do that, in effect keeping the "flag" world separate from the "version" world.

I haven't changed `deploymentsByEnvironmentName`, but that would be possible.  I suspect that one of the issues here is that this information is sourced from _deployments-service_ which doesn't know anything about `SlugInfoFlag` (which is a _dependencies-service_ concept).  As you can see from this function in `CatalogueController` there is a lot of data being pulled from disparate sources here - and then much of the assembly is being done in the page itself.  There is quite a lot of complexity here, and it would be nice to spend some time abstracting over the various services and building a model for the view to simply display.